src/Controller/EventsController.php

```diff
@@ -733,16 +734,25 @@ public function rejectAction(Request $request, PDO $db)
 
         $event_id     = $this->getItemId($request);
         $event_mapper = new EventMapper($db, $request);
+        $reason = filter_var($request->getParameter('reason'), FILTER_SANITIZE_STRING);
 
         if (!$event_mapper->thisUserCanApproveEvents()) {
             throw new Exception("You are not allowed to reject this event", Http::FORBIDDEN);
         }
 
-        $result = $event_mapper->reject($event_id, $request->user_id);
+        $result = $event_mapper->reject($event_id, $request->user_id, $reason);
         if (!$result) {
             throw new Exception("This event cannot be rejected", Http::BAD_REQUEST);
         }
 
+        // Only send an email if we provided a reason
+        if (!empty($reason)) {
+            $event        = $event_mapper->getEventById($event_id, true, false)['events'][0];
```

Comment:
Can we even be sure that we have an event for `$event_id` here, that is, blindly assume that the return value of `getEventById()` will be an array that has an `events` key, where the corresponding value is an array that that has an entry for key `0`?

Reply:
I agree with your concerns but we've not run into issues that I'm aware of. (No one has reported issues rejecting events, and there have certainly been a few rejected over the past few weeks.)